### PR TITLE
DEPR: Remove EdxRestApiClient usage in tubular

### PR DESCRIPTION
**Pull request by ruzniaievdm**
YT: https://youtrack.raccoongang.com/issue/OeX_Depr-36
JIRA: https://openedx.atlassian.net/browse/DEPR-138

### CODE REVIEW

**Review by dyudyunov — APPROVED** [review]

File: tubular/edx_api.py [hunk 1]
```diff
@@ -29,46 +32,101 @@ class BaseApiClient:
...
-        access_token, __ = self.get_access_token(lms_base_url, client_id, client_secret)
-        self.create_client(access_token)
+        self._access_token = self.get_access_token(lms_base_url, client_id, client_secret)
 
-    def create_client(self, access_token):
+    def get_api_url(self, path):
         """
-        Creates and stores the EdxRestApiClient that we use to actually make requests.
+        Construct the full API URL using the api_base_url and path.
+        Args:
```
Comment (idegtiarov) [inline tubular/edx_api.py:48]: Please add an empty line between the title and the doc string body

File: tubular/edx_api.py [hunk 2]
```diff
@@ -29,46 +32,101 @@ class BaseApiClient:
...
+            if status_code == 504:
+                # Differentiate gateway errors so different backoff can be used.
+                raise EdxGatewayTimeoutError(str(exc))
+
+            if status_code == 404:
+                raise HttpDoesNotExistException(str(exc))
+            raise
+
+        except Timeout:
+            LOG.error("The request timed out.")
```
Comment (idegtiarov) [inline tubular/edx_api.py]: typo: s/The request timed out/The request is timed out/

File: tubular/edx_api.py [hunk 3]
```diff
@@ -29,46 +32,101 @@ class BaseApiClient:
...
             )
-        except HttpClientError as err:
-            LOG.error("API Error: {} with status code: {} fetching access token for client: {}".format(
-                err.content,
-                err.response.status_code,  # pylint: disable=no-member
+            response.raise_for_status()
+            data = response.json()
+            access_token = data['access_token']
+        except KeyError as exc:
+            LOG.error("Failed to get token. {} does not exist.".format(str(exc)))
```
Comment (idegtiarov) [inline tubular/edx_api.py]: I would use f-string instead of format here and below, but it is on you.

File: tubular/edx_api.py [hunk 4]
```diff
@@ -101,14 +159,15 @@ def _giveup_on_unexpected_exception(exc):
     )
     return not keep_retrying
 
+
```
Comment (idegtiarov) [inline tubular/edx_api.py:158]: weird change, why?

File: tubular/edx_api.py [hunk 5]
```diff
@@ -173,11 +199,11 @@ def learners_to_retire(self, states_to_request, cool_off_days=7):
         Retrieves a list of learners awaiting retirement actions.
         """
         params = {
+            'states': states_to_request,
             'cool_off_days': cool_off_days,
-            'states': states_to_request
```
Comment (idegtiarov) [inline tubular/edx_api.py:177]: some weird change here as well

File: tubular/scripts/retire_one_learner.py [hunk 6]
```diff
@@ -4,6 +4,7 @@
 
 To run this script you will need a username to run against and a YAML config file in the format:
 
+
```
Comment (idegtiarov) [inline tubular/scripts/retire_one_learner.py]: do we need that additional space?

File: tubular/scripts/retire_one_learner.py [hunk 7]
```diff
@@ -22,29 +23,23 @@
...
 # pylint: disable=wrong-import-position
-from tubular.scripts.helpers import (
-    _config_or_exit,
-    _fail,
-    _fail_exception,
-    _get_error_str_from_exception,
-    _log,
-    _setup_all_apis_or_exit
-)
+from tubular.scripts.helpers import (_config_or_exit, _fail, _fail_exception,
```
Comment (idegtiarov) [inline tubular/scripts/retire_one_learner.py]: this change is also not looked to be correct. Could it be reverted to the previous approach?

File: tubular/tests/test_get_learners_to_retire.py [hunk 8]
```diff
@@ -99,7 +99,7 @@ def test_misconfigured(*args, **kwargs):
     mock_get_learners_to_retire = kwargs['learners_to_retire']
 
     mock_get_access_token.return_value = ('THIS_IS_A_JWT', None)
-    mock_get_learners_to_retire.side_effect = HttpClientError
+    mock_get_learners_to_retire.side_effect = RequestException
```
Comment (idegtiarov) [inline tubular/tests/test_get_learners_to_retire.py]: Please clarify why we do not specify the exception in the side_effect but use a generic one?

File: tubular/tests/test_retire_one_learner.py [hunk 9]
```diff
@@ -2,22 +2,19 @@
...
-    ERR_UNKNOWN_STATE,
-    ERR_USER_AT_END_STATE,
-    ERR_USER_IN_WORKING_STATE,
-    retire_learner
-)
-from tubular.tests.retirement_helpers import fake_config_file, get_fake_user_retirement
+from mock import DEFAULT, patch
+
+from tubular.exception import HttpDoesNotExistException
+from tubular.scripts.retire_one_learner import (END_STATES, ERR_BAD_CONFIG,
```
Comment (idegtiarov) [inline tubular/tests/test_retire_one_learner.py]: long line splitting is recommended to be done differently please take a look at a python style guide https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines

File: tubular/tests/test_retire_one_learner.py [hunk 10]
```diff
@@ -2,22 +2,19 @@
...
+                                                ERR_BAD_LEARNER,
+                                                ERR_SETUP_FAILED,
+                                                ERR_UNKNOWN_STATE,
+                                                ERR_USER_AT_END_STATE,
+                                                ERR_USER_IN_WORKING_STATE,
+                                                retire_learner)
+from tubular.tests.retirement_helpers import (fake_config_file,
```
Comment (idegtiarov) [inline tubular/tests/test_retire_one_learner.py]: ditto

**Review by idegtiarov — CHANGES_REQUESTED** [review]
Thank you for the new PR! 
Some comments, questions, and recommendations are inline, please take a look.

File: tubular/edx_api.py (hunk 4 above)
Comment (ruzniaievdm) [reply tubular/edx_api.py:158]: added extra indent for double indentation between methods outside classes

File: tubular/edx_api.py [hunk 11]
```diff
@@ -199,8 +198,8 @@ def learners_to_retire(self, states_to_request, cool_off_days=7):
         Retrieves a list of learners awaiting retirement actions.
         """
         params = {
-            'states': states_to_request,
             'cool_off_days': cool_off_days,
+            'states': states_to_request
```
Comment (idegtiarov) [inline tubular/edx_api.py:199]: this change doesn't look meaningful, it should be reverted

File: tubular/tests/test_edx_api.py [hunk 12]
```diff
@@ -96,15 +318,244 @@ def test_retirement_partner_cleanup_backoff_on_connection_error(
             mock_retirement_partner_cleanup
     ):
         mock_backoff_handler.side_effect = BackoffTriedException
+        response = requests.Response()
+        response.status_code = svr_status_code
+        mock_retirement_partner_cleanup.retirement_partner_cleanup.side_effect = ConnectionError(
+            response=response)
```
Comment (idegtiarov) [inline tubular/tests/test_edx_api.py]: please put trailing parenthesis down to the new line

**Review by idegtiarov — CHANGES_REQUESTED** [review]
Thank you for the PR, I've left comments for two minor nits other LGTM!

File: tubular/tests/test_edx_api.py (hunk 12 above)
Comment (ruzniaievdm) [reply tubular/tests/test_edx_api.py]: Done

File: tubular/edx_api.py (hunk 11 above)
Comment (ruzniaievdm) [reply tubular/edx_api.py:199]: thanks, I fixed that in updated pr to upstream after rebasing

**Review by idegtiarov — APPROVED** [review]
Thank you please open PR to upstream!

File: tubular/edx_api.py [hunk 13]
```diff
@@ -29,46 +32,100 @@ class BaseApiClient:
...
             )
-        except HttpClientError as err:
-            LOG.error("API Error: {} with status code: {} fetching access token for client: {}".format(
-                err.content,
-                err.response.status_code,  # pylint: disable=no-member
-                client_id,
-            ))
+            response.raise_for_status()
+            data = response.json()
+            access_token = data['access_token']
```
Comment (dyudyunov) [inline tubular/edx_api.py]: you can move the return statement from line 127 here:
`return response.json().get('access_tokent')`

File: tubular/tests/retirement_helpers.py [hunk 14]
```diff
@@ -5,19 +5,32 @@
...
+from datetime import datetime
 
 import yaml
 
 
 TEST_RETIREMENT_PIPELINE = [
     ['RETIRING_FORUMS', 'FORUMS_COMPLETE', 'LMS', 'retirement_retire_forum'],
-    ['RETIRING_EMAIL_LISTS', 'EMAIL_LISTS_COMPLETE', 'LMS', 'retirement_retire_mailings'],
+    ['RETIRING_EMAIL_LISTS', 'EMAIL_LISTS_COMPLETE',
+        'LMS', 'retirement_retire_mailings'],
```
Comment (dyudyunov) [inline tubular/tests/retirement_helpers.py]: why do you need this change?

**Review by dyudyunov — APPROVED** [review]
Cool tests! 💪

Few minor comments inline, the other - LGTM!

File: tubular/tests/retirement_helpers.py (hunk 14 above)
Comment (ruzniaievdm) [reply tubular/tests/retirement_helpers.py]: yes, this is redundant, I already remove it changes, after rebasing I will update

File: tubular/edx_api.py (hunk 13 above)
Comment (ruzniaievdm) [reply tubular/edx_api.py]: added this changes, thanks

**Pull request closed without merging**